src/widget.cpp

```diff
@@ -545,10 +549,10 @@ void widget::finalize()
     widget_factory.finalize();
 
     for( const widget &wgt : widget::get_all() ) {
-        if( wgt.explicit_separator || wgt.explicit_padding ) {
+        if( wgt._style == "sidebar" ) {
```

Comment:
The idea here was to propagate separators and padding down from widgets that explicitly defined it so we didn't need to redefine it everywhere in JSON. Does it still work that way? It doesn't look like it.

Reply:
Starting up the game, it looks like my tired brain forgot how I implemented `finalize_label_width_recursive`, and things still work just fine. Again - Marvelous work.

Reply:
Thanks!

Yes, still works I hope. The change just prevents it from traversing partial trees multiple times. Which could also have resulted in unwanted overwrites, depending on load order. Now, it should be consistent independent of the order of definition.